File: dp_primer/lis_continuous_step0/cs/Program.cs
/*
これはpaizaラーニングの問題集にある「DPメニュー」-「最長増加連続部分列」
https://paiza.jp/works/mondai/dp_primer/dp_primer_lis_continuous_step0
にC#でチャレンジしたコードです。

作成環境
Windows 10 Pro
Microsoft Visual Studio Community 2019 (.NET 5.0)
*/
using System;
using System.Linq;

class Program
{
    static void Main()
    {
        // 人数 n
        var line1 = Console.ReadLine().Trim();
        var n = int.Parse(line1);
        // 身長 a_n
        var a = new int[n];
        for (var i = 0; i < n; i++)
        {
            var line2 = Console.ReadLine().Trim();
            a[i] = int.Parse(line2);
        }
        // 動的計画法で問題を解く
        var dp = new int[n];
        dp[0] = 1;
        for (var i = 1; i < n; i++)
        {
            if (a[i-1] <= a[i])
            {
                dp[i] = dp[i - 1] + 1;
            } else
            {
                dp[i] = 1;
            }
        }
        // 結果出力
        Console.WriteLine(dp.Max());
    }
}

File: dp_primer/lis_continuous_step0/vb/Program.vb
' これはpaizaラーニングの問題集にある「DPメニュー」-「最長増加連続部分列」
' https://paiza.jp/works/mondai/dp_primer/dp_primer_lis_continuous_step0
' にVisual Basicでチャレンジしたコードです。
'
' 作成環境
' Windows 10 Pro
' Microsoft Visual Studio Community 2019 (.NET 5.0)

Imports System

Module Program
    Sub Main()
        ' 人数 n
        Dim line1 As String = Console.ReadLine().Trim()
        Dim n As Integer = Integer.Parse(line1)
        ' 身長 a_n
        Dim a(n) As Integer
        For i As Integer = 0 To n - 1
            Dim line2 As String = Console.ReadLine().Trim()
            a(i) = Integer.Parse(line2)
        Next
        ' 動的計画法で問題を解く
        Dim dp(n) As Integer
        dp(0) = 1
        For i As Integer = 1 To n - 1
            If a(i - 1) <= a(i) Then
                dp(i) = dp(i - 1) + 1
            Else
                dp(i) = 1
            End If
        Next
        ' 結果出力
        Dim max As Integer = dp(0)
        For i As Integer = 1 To n - 1
            If max < dp(i) Then
                max = dp(i)
            End If
        Next
        Console.WriteLine(max)
    End Sub
End Module
